[RDBMS] Add support for advanced threat protection setting for a PostgreSQL flexible server

ambrahma opened the pull request:
**Related command**
`az postgres flexible-server advanced-threat-protection-setting update --resource-group testGroup --server-name testserver --state Enabled/Disabled' to enable/disable state for advanced threat protection setting.
`az postgres flexible-server advanced-threat-protection-setting show --resource-group testGroup --server-name testserver` to show the advanced server threat protection setting

**Description**<!--Mandatory-->
Added support for Advanced Server Threat Protection Setting for PostgreSQL flexible server. It supports commands to update and show the advanced threat protection setting

**Testing Guide**
Added automated tests

**History Notes**
<!--If your PR is not customer-facing, use {Component Name} in the PR title. Otherwise, use [Component Name] to allow our pipeline to add the title as a history note. If you need multiple history notes or would like to overwrite the note from the PR title, please fill in the following templates.-->

[RDBMS] `az postgres flexible-server advanced-threat-protection-setting show`: Show advanced threat protection setting
[RDBMS] `az postgres flexible-server advanced-threat-protection-setting update`: Update advanced threat protection setting using `--state` as Enabled/Disabled

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [x] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

yonzhan commented:
<!-- add-pr-comment:add-pr-comment -->

RDBMS

Ran-br commented on src/azure-cli/azure/cli/command_modules/rdbms/_params.py:
```diff
@@ -809,6 +809,20 @@ def _flexible_server_params(command_group):
+        for scope in ['update', 'show']:
+            argument_context_string = '{} flexible-server advanced-threat-protection-setting {}'.format(command_group, scope)
+            with self.argument_context(argument_context_string) as c:
+                c.argument('resource_group_name', arg_type=resource_group_name_type)
```
Is this only applicable in the context of the current subscription? (no subscription param)
Also, on az sql there's an option to specify '--ids' which enables some sort of bulk operations by calling show/ update on multiple resource ids: [az-sql-server-advanced-threat-protection-setting](https://learn.microsoft.com/en-us/cli/azure/sql/server/advanced-threat-protection-setting?view=azure-cli-latest#az-sql-server-advanced-threat-protection-setting-update:~:text=default%20value%3A%20False-,%2D%2Dids,-One%20or%20more)

ambrahma replied:
Updated help

evelyn-ys commented on src/azure-cli/azure/cli/command_modules/rdbms/_helptext_pg.py:
```diff
@@ -746,3 +746,34 @@
+
+helps['postgres flexible-server advanced-threat-protection-setting update'] = """
+type: command
+short-summary: Updates advanced threat protection setting state for a flexible server.
```
```suggestion
short-summary: Update advanced threat protection setting state for a flexible server.
```

evelyn-ys commented on src/azure-cli/azure/cli/command_modules/rdbms/flexible_server_commands.py:
```diff
@@ -184,3 +190,9 @@ def load_flexibleserver_command_table(self, _):
+    with self.command_group('postgres flexible-server advanced-threat-protection-setting', postgres_flexible_server_threat_protection_settings_sdk,
+                            custom_command_type=flexible_servers_custom_postgres,
+                            client_factory=cf_postgres_flexible_server_threat_protection_settings) as g:
+        g.custom_command('show', 'flexible_server_threat_protection_get', custom_command_type=flexible_servers_custom_postgres)
```
```suggestion
        g.custom_show_command('show', 'flexible_server_threat_protection_get', custom_command_type=flexible_servers_custom_postgres)
```

ambrahma replied:
Thank you, this solved the issue.

evelyn-ys commented:
Kind reminder that our code complete date is 11/28/2023 10:00 UTC. Please address comments and CI failures in time if you want to catch [November 2023 (2023-12-05)](https://github.com/Azure/azure-cli/milestone/136)

ambrahma commented:
> Kind reminder that our code complete date is 11/28/2023 10:00 UTC. Please address comments and CI failures in time if you want to catch [November 2023 (2023-12-05)](https://github.com/Azure/azure-cli/milestone/136)

I have addressed comments and CI failures. PR is ready for merge
